src/common/ZulipTextButton.js

```diff
@@ -98,21 +105,37 @@ type Props = $ReadOnly<{|
 // things like project-specific styles and making any sensible adjustments
 // to the interface.
 export default function ZulipTextButton(props: Props): Node {
-  const { variant = 'standard', leftMargin, rightMargin, label, onPress } = props;
+  const { variant = 'standard', leftMargin, rightMargin, label, onPress, disabled } = props;
 
   const variantStyles = useMemo(() => styleSheetForVariant(variant), [variant]);
 
+  const isNotDisabled = disabled !== true;
+
+  const disabledGate = useCallback(
+    onPressCallback => () => {
+      if (isNotDisabled) {
+        onPressCallback();
+      }
+    },
+    [isNotDisabled],
+  );
+
+  const buttonDisabledColor = useMemo(
+    () => ({ color: isNotDisabled ? BRAND_COLOR : HALF_COLOR }),
+    [isNotDisabled],
+  );
```

Comment:
I'm wondering if there should be a ternary conditional inside `useMemo`. In retrospect, the call should just memoize a value provided, without it having to do further operations.

Reply:
```js
const color = isNotDisabled ? BRAND_COLOR : HALF_COLOR;

const buttonDisabledColor = useMemo(
  () => ({ color }),
  [color],
);
```

Reply:
I'm beginning to think the performance cost isn't worth it. This [blog shows devs testing](https://canimerge.com/should-you-really-use-usememo-in-react-lets-find-out/) the overhead cost `useMemo` against performance gain.

Reply:
I really appreciate the deep dive. 🔍 I didn't know what explicitly the variant function was for so I thought we might want disabled regardless of variant. Including it in `styleSheetForVariant` is much cleaner, though. 

I think the mistake I made was thinking that the ternary conditional statement isn't evaluated before it's consumed.